bitcoin/src/blockdata/script/witness_program.rs

```diff
@@ -99,7 +115,12 @@ impl WitnessProgram {
     pub fn version(&self) -> WitnessVersion { self.version }
 
     /// Returns the witness program.
-    pub fn program(&self) -> &PushBytes { &self.program }
+    pub fn program(&self) -> &PushBytes {
+        self.program
+            .as_slice()
+            .try_into()
+            .expect("witness programs are always smaller than max size of PushBytes")
+    }
```

Comment:
FTR, this is why I wanted `WitnessProgram` to not contain version and then just impl `AsPushBytes` for it... Anyway, this is good now, we will need a lot of refactoring to make it perfect.